src/core/index.ts

```diff
@@ -13,7 +13,7 @@ import type { ErrorType } from "./pages/error"
 export interface IncomingRequest {
   /** @default "http://localhost:3000" */
   host: string
-  method: string
+  method?: string
```

Comment:
I think we could type this more strictly?
```suggestion
  method?: 'GET' | 'POST' | 'PUT' | 'DELETE' | 'PATCH' | 'HEAD' | 'OPTIONS' | 'TRACE'
```

Reply:
I don't see an immediate value in it. The core handler only handles `"GET" | "POST"` anyway, otherwise, it will fall back to generate a 400 bad request response at the bottom.

I checked `Request["method"]` for reference, if that defines this type as `string`:

![image](https://user-images.githubusercontent.com/18369201/142287582-32095f1b-a69a-45dc-b365-084252ea3676.png)

If we find a real use case/confusion, we can address this in a different PR.